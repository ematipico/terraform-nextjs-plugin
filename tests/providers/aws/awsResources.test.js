// @ts-ignore
const AwsResources = require("../../../src/providers/aws");
const AwsConfig = require("../../../src/providers/aws/awsConfig");
const path = require("path");

describe("AwsResources", () => {
	it("should generate the correct configuration", async () => {
		const config = new AwsConfig({
			gatewayKey: "CustomKey",
			// eslint-disable-next-line unicorn/prevent-abbreviations
			nextAppDir: path.resolve(__dirname, "__fixtures__"),
			provider: "AWS",
			routes: {
				prefix: "",
				mappings: [
					{
						page: "/blog-post",
						route: "/new-blog-post-amazing"
					},
					{
						page: "/singleBlogPost",
						route: "/blog/:url"
					},
					{
						page: "/blogPost",
						route: "/blog/detail/:url",
						params: {
							page: true,
							hideComments: false
						}
					}
				]
			}
		});

		const awsResources = new AwsResources(config);
		const result = await awsResources.generateGatewayResources();

		expect(result).toMatchSnapshot();
	});
});
