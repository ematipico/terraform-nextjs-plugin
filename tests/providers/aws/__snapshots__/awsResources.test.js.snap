// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AwsResources should generate the correct configuration 1`] = `
Object {
  "resource": Object {
    "aws_api_gateway_integration": Object {
      "CustomKey-": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "request_parameters": Object {
          "method.request.querystring.hideComments": false,
          "method.request.querystring.page": true,
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-detail": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "request_parameters": Object {
          "method.request.path.url": true,
          "method.request.querystring.hideComments": false,
          "method.request.querystring.page": true,
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog-detail.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
    },
    "aws_api_gateway_method": Object {
      "CustomKey-": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "resource_id": "\${aws_api_gateway_resource.CustomKey-.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "request_parameters": Object {
          "method.request.querystring.hideComments": false,
          "method.request.querystring.page": true,
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-detail": Object {
        "authorization": "NONE",
        "http_method": "GET",
        "request_parameters": Object {
          "method.request.path.url": true,
          "method.request.querystring.hideComments": false,
          "method.request.querystring.page": true,
        },
        "resource_id": "\${aws_api_gateway_resource.CustomKey-blog-detail.id}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
    },
    "aws_api_gateway_resource": Object {
      "CustomKey-": Object {
        "parent_id": "\${aws_api_gateway_rest_api.CustomKey.root_resource_id}",
        "path_part": "index",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog": Object {
        "parent_id": "\${aws_api_gateway_resource.blog.id}",
        "path_part": "detail",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
      "CustomKey-blog-detail": Object {
        "parent_id": "\${aws_api_gateway_resource.blog-detail.id}",
        "path_part": "{url}",
        "rest_api_id": "\${aws_api_gateway_rest_api.CustomKey.id}",
      },
    },
  },
  "variable": Object {
    "integrationList": Object {
      "default": Array [
        "aws_api_gateway_integration.CustomKey-",
        "aws_api_gateway_integration.CustomKey-blog",
        "aws_api_gateway_integration.CustomKey-blog-detail",
      ],
    },
  },
}
`;
